src/ripple/rpc/handlers/AccountChannels.cpp

```diff
@@ -40,9 +40,12 @@ void addChannel (Json::Value& jsonLines, SLE const& line)
     jDst[jss::destination_account] = to_string (line[sfDestination]);
     jDst[jss::amount] = line[sfAmount].getText ();
     jDst[jss::balance] = line[sfBalance].getText ();
-    PublicKey const pk (line[sfPublicKey]);
-    jDst[jss::public_key] = toBase58 (TokenType::TOKEN_ACCOUNT_PUBLIC, pk);
-    jDst[jss::public_key_hex] = strHex (pk);
+    if (publicKeyType(line[sfPublicKey]))
+    {
+        PublicKey const pk (line[sfPublicKey]);
+        jDst[jss::public_key] = toBase58 (TokenType::TOKEN_ACCOUNT_PUBLIC, pk);
+        jDst[jss::public_key_hex] = strHex (pk);
+    }
```

Comment:
sorry - discarded is a poor choice of words -- we ignore whatever data is in `line[sfPublicKey]` by not setting the corresponding `jDst` fields when the check fails. This is probably the right thing to do..I was just asking for my own clarification and I was wondering if some sort of log message or user warning made sense. If it's SOP to ignore malformed fields, then this makes complete sense as is.